src/librustc_mir/util/liveness.rs

```diff
@@ -138,7 +138,7 @@ impl<V: LiveVariableMap> LivenessResults<V> {
 /// Compute which local variables are live within the given function
 /// `mir`. The liveness mode `mode` determines what sorts of uses are
 /// considered to make a variable live (e.g., do drops count?).
-pub fn liveness_of_locals<'tcx, V: LiveVariableMap>(mir: &Mir<'tcx>, mode: LivenessMode) -> LivenessResult<V::LiveVar> {
+pub fn liveness_of_locals<'tcx, V>(mir: &Mir<'tcx>, mode: LivenessMode) -> LivenessResult<V> {
     let locals = mir.local_decls.len();
```

Comment:
this variable represents the total number of variables we are computing liveness for. We may need to extend the `LiveVariableMap` trait to include a `num_variables(mir: &Mir<'tcx>)` method -- for the identity case, it would return `mir.local_decls.len()`.